Source/WebCore/page/Page.cpp

```diff
@@ -4236,6 +4236,37 @@ void Page::forEachWindowEventLoop(const Function<void(WindowEventLoop&)>& functo
         functor(eventLoop);
 }
 
+RefPtr<Element> Page::elementForContext(const ElementContext& elementContext) const
+{
+    if (!m_identifier || elementContext.webPageIdentifier != *m_identifier)
+        return nullptr;
+
+    RefPtr element = elementContext.elementIdentifier ? Element::fromIdentifier(*elementContext.elementIdentifier) : nullptr;
+    if (!element)
+        return nullptr;
+
+    if (!element->isConnected() || element->document().identifier() != elementContext.documentIdentifier || element->document().page() != this)
+        return nullptr;
+
+    return element;
+}
+
+std::optional<ElementContext> Page::contextForElement(const Element& element) const
+{
+    if (!m_identifier)
+        return std::nullopt;
+
+    Ref document = element.document();
+    if (document->page() != this)
+        return std::nullopt;
+
+    RefPtr frame = document->frame();
+    if (!frame)
+        return std::nullopt;
+
+    return ElementContext { element.boundingBoxInRootViewCoordinates(), *m_identifier, document->identifier(), element.identifier() };
```

Comment:
Do all clients need the bounding box? That's quite expensive to compute.

Reply:
This was copied from `WebPage::contextForElement()`. In the new revision of the patch, I'm going to introduce a new `HostedModel` in `WebCore/platform` that has the minimum information we need for now. And this change will be reverted.